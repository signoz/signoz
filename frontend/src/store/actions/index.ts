export * from './types';
export * from './traceFilters';
export * from './serviceMap';
export * from './traces';
export * from './MetricsActions';
export * from './usage';
export * from './global';
