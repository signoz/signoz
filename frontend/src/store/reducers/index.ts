import { combineReducers } from 'redux';
import {
	traceResponseNew,
	spansWSameTraceIDResponse,
	usageDataItem,
	GlobalTime,
	serviceMapStore,
	TraceFilters,
} from '../actions';
import { updateGlobalTimeReducer } from './global';
import { MetricsInitialState, metricsReducer } from './metrics';
import TraceFilterReducer from './traceFilters';
import { traceItemReducer, tracesReducer } from './traces';
import { usageDataReducer } from './usage';
import { ServiceMapReducer } from './serviceMap';

export interface StoreState {
	metricsData: MetricsInitialState;
	traceFilters: TraceFilters;
	traces: traceResponseNew;
	traceItem: spansWSameTraceIDResponse;
	usageDate: usageDataItem[];
	globalTime: GlobalTime;
	serviceMap: serviceMapStore;
}

const reducers = combineReducers<StoreState>({
	traceFilters: TraceFilterReducer,
	traces: tracesReducer,
	traceItem: traceItemReducer,
	usageDate: usageDataReducer,
	globalTime: updateGlobalTimeReducer,
	metricsData: metricsReducer,
	serviceMap: ServiceMapReducer,
});

export default reducers;
