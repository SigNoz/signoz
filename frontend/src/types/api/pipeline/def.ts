import { TagFilter } from '../queryBuilder/queryBuilderData';

export interface ProcessorData {
	type: string;
	id?: string;
	orderId: number;
	name: string;
	enabled?: boolean;
	output?: string;
	parse_to?: string;
	pattern?: string;
	parse_from?: string;
	from?: string;
	to?: string;
	regex?: string;
	on_error?: string;
	field?: string;
	value?: string;

	// trace parser fields.
	trace_id?: {
		parse_from: string;
	};
	span_id?: {
		parse_from: string;
	};
	trace_flags?: {
		parse_from: string;
	};

	// time parser fields
	layout_type?: string;
	layout?: string;

	// json flattening fields
	enable_flattening?: boolean;
	enable_paths?: boolean;
	path_prefix?: string;
	enable_mapping?: boolean;
	mapping?: Record<string, string[]>;
}

export interface PipelineData {
	alias: string;
	config?: Array<ProcessorData>;
	createdAt: string;
	description?: string;
	createdBy: string;
	enabled: boolean;
	filter: TagFilter;
	id?: string;
	name: string;
	orderId: number;
	tags?: Array<string>; // Tags data is missing in API response
}

export interface HistoryData {
	active: boolean;
	createdAt: string;
	createdBy: string;
	createdByName: string;
	deployStatus: string;
	deployResult: string;
	disabled: boolean;
	elementType: string;
	id: string;
	isValid: boolean;
	lastConf: string;
	lastHash: string;
	version: number;
}

export interface Pipeline {
	active: boolean;
	createdBy: string;
	deployResult: string;
	deployStatus: string;
	disabled: boolean;
	elementType: string;
	history: Array<HistoryData>;
	id: string;
	is_valid: boolean;
	lastConf: string;
	lastHash: string;
	pipelines: Array<PipelineData>;
	version: string | number;
}

export enum ActionType {
	AddPipeline = 'add-pipeline',
	EditPipeline = 'edit-pipeline',
	AddProcessor = 'add-processor',
	EditProcessor = 'edit-processor',
}

export enum ActionMode {
	Viewing = 'viewing-mode',
	Editing = 'editing-mode',
	Deploying = 'deploying-mode',
}
