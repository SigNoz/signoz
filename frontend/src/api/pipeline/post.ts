import axios from 'api';
import { ErrorResponseHandlerV2 } from 'api/ErrorResponseHandlerV2';
import { AxiosError } from 'axios';
import { ErrorV2Resp, SuccessResponseV2 } from 'types/api';
import { Pipeline } from 'types/api/pipeline/def';
import { Props } from 'types/api/pipeline/post';

const post = async (props: Props): Promise<SuccessResponseV2<Pipeline>> => {
	try {
		const response = await axios.post('/logs/pipelines', props.data);

		return {
			httpStatusCode: response.status,
			data: response.data.data,
		};
	} catch (error) {
		ErrorResponseHandlerV2(error as AxiosError<ErrorV2Resp>);
	}
};

export default post;
