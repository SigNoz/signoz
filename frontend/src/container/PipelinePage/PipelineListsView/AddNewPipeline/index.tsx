import { Button, Divider, Form, Modal } from 'antd';
import { useAppContext } from 'providers/App/App';
import React, { useCallback, useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { ActionMode, ActionType, PipelineData } from 'types/api/pipeline/def';
import { v4 } from 'uuid';

import { ModalButtonWrapper, ModalTitle } from '../styles';
import { getEditedDataSource, getRecordIndex } from '../utils';
import { renderPipelineForm } from './utils';

function AddNewPipeline({
	isActionType,
	setActionType,
	selectedPipelineData,
	setShowSaveButton,
	setCurrPipelineData,
	currPipelineData,
}: AddNewPipelineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation('pipeline');
	const { user } = useAppContext();

	const isEdit = isActionType === 'edit-pipeline';
	const isAdd = isActionType === 'add-pipeline';

	useEffect(() => {
		if (isEdit) {
			form.setFieldsValue(selectedPipelineData);
		}
		if (isAdd) {
			form.resetFields();
		}
	}, [form, isEdit, isAdd, selectedPipelineData]);

	const onFinish = (values: PipelineData): void => {
		const newPipeLineData: PipelineData = {
			id: v4(),
			orderId: (currPipelineData?.length || 0) + 1,
			createdAt: new Date().toISOString(),
			createdBy: user?.name || '',
			name: values.name,
			alias: values.name.replace(/\s/g, ''),
			description: values.description,
			filter: values.filter,
			config: [],
			enabled: true,
		};

		if (isEdit && selectedPipelineData) {
			const findRecordIndex = getRecordIndex(
				currPipelineData,
				selectedPipelineData,
				'id',
			);
			const updatedPipelineData: PipelineData = {
				...currPipelineData[findRecordIndex],
				...values,
			};

			const editedPipelineData = getEditedDataSource(
				currPipelineData,
				selectedPipelineData,
				'id',
				updatedPipelineData,
			);

			setCurrPipelineData(editedPipelineData);
		}
		if (isAdd) {
			setCurrPipelineData((prevState) => {
				if (prevState) return [...prevState, newPipeLineData];
				return [newPipeLineData];
			});
		}
		setActionType(undefined);
	};

	const onCancelModalHandler = (): void => {
		setActionType(undefined);
	};

	const modalTitle = useMemo(
		(): string =>
			isEdit
				? `${t('edit_pipeline')} : ${selectedPipelineData?.name}`
				: t('create_pipeline'),
		[isEdit, selectedPipelineData?.name, t],
	);

	const onOkModalHandler = useCallback(
		() => setShowSaveButton(ActionMode.Editing),
		[setShowSaveButton],
	);

	const isOpen = useMemo(() => isEdit || isAdd, [isAdd, isEdit]);

	return (
		<Modal
			title={<ModalTitle level={4}>{modalTitle}</ModalTitle>}
			centered
			open={isOpen}
			width={800}
			footer={null}
			onCancel={onCancelModalHandler}
		>
			<Divider plain />
			<Form
				name="add-new-pipeline"
				layout="vertical"
				onFinish={onFinish}
				autoComplete="off"
				form={form}
			>
				{renderPipelineForm()}
				<Divider plain />
				<Form.Item>
					<ModalButtonWrapper>
						<Button
							key="submit"
							type="primary"
							htmlType="submit"
							onClick={onOkModalHandler}
						>
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={onCancelModalHandler}>
							{t('cancel')}
						</Button>
					</ModalButtonWrapper>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPipelineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedPipelineData: PipelineData | undefined;
	setShowSaveButton: (actionMode: ActionMode) => void;
	setCurrPipelineData: (
		value: React.SetStateAction<Array<PipelineData>>,
	) => void;
	currPipelineData: Array<PipelineData>;
}

export default AddNewPipeline;
