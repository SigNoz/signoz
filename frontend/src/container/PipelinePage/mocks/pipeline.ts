import { Pipeline, PipelineData } from 'types/api/pipeline/def';
import { DataTypes } from 'types/api/queryBuilder/queryAutocompleteResponse';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';

export const configurationVersion = '1.0';

export function mockPipelineFilter(
	key: string,
	op: string,
	value: string,
): TagFilter {
	return {
		op: 'AND',
		items: [
			{
				id: `${key}-${value}`,
				key: {
					key,
					dataType: DataTypes.String,
					type: '',
					isColumn: false,
					isJSON: false,
				},
				op,
				value,
			},
		],
	};
}

export const pipelineMockData: Array<PipelineData> = [
	{
		id: '4453c8b0-c0fd-42bf-bf09-7cc1b04ccdc9',
		orderId: 1,
		name: 'Apache common parser',
		alias: 'apachecommonparser',
		description: 'This is a desc',
		enabled: false,
		filter: mockPipelineFilter('source', '=', 'nginx'),
		config: [
			{
				orderId: 1,
				enabled: true,
				type: 'grok_parser',
				id: 'grokusecommon',
				name: 'grok use common asd',
				output: 'renameauth',
				parse_to: 'attributes',
				pattern: '%{COMMONAPACHELOG}',
				parse_from: 'body',
			},
			{
				orderId: 2,
				enabled: true,
				type: 'move',
				id: 'renameauth',
				name: 'rename auth',
				from: 'attributes.auth',
				to: 'attributes.username',
			},
			{
				orderId: 3,
				enabled: true,
				type: 'json_parser',
				id: 'jsonparser',
				name: 'json parser',
				from: 'attributes.auth',
				to: 'attributes.username',
			},
		],
		createdBy: 'nityananda@signoz.io',
		createdAt: '2023-03-07T16:56:53.36071141Z',
	},
	{
		id: 'a3675a0c-ff73-4ddb-be39-4351ace69231',
		orderId: 2,
		name: 'Moving pipeline new',
		alias: 'movingpipelinenew',
		description: 'This is a desc of move',
		enabled: false,
		filter: mockPipelineFilter('method', '=', 'POST'),
		config: [
			{
				orderId: 1,
				enabled: true,
				type: 'copy',
				id: 'mv1',
				name: 'mymove',
				from: 'attributes.method',
				to: 'attributes.method11',
			},
		],
		createdBy: 'chintan@signoz.io',
		createdAt: '2023-03-07T16:55:27.789595116Z',
	},
];

export const pipelineApiResponseMockData: Pipeline = {
	id: '67ace08a-6b6c-4221-ab58-a5d3bd5eb6f2',
	version: 5,
	elementType: 'log_pipelines',
	createdBy: 'test@signoz.io',
	active: false,
	is_valid: false,
	disabled: false,
	deployStatus: 'IN_PROGRESS',
	deployResult: 'Deployment started',
	lastHash: 'q��҂�&覣ʝup��\u0003���q�6�\u001e�<ѥIb�',
	lastConf:
		'[{"id":"a3675a0c-ff73-4ddb-be39-4351ace69231","orderId":"1","name":"Moving pipeline new","alias":"movingpipelinenew","enabled":false,"filter":"attributes.method == POST","config":[{"type":"copy","id":"mv1","name":"mymove","from":"attributes.method","to":"attributes.method11"}],"createdBy":"nityananda@signoz.io","createdAt":"2023-03-07T16:55:27.789595116Z"},{"id":"4453c8b0-c0fd-42bf-bf09-7cc1b04ccdc9","orderId":"2","name":"Apache common parser","alias":"apachecommonparser","enabled":false,"filter":"attributes.source == nginx","config":[{"type":"grok_parser","id":"grokusecommon","name":"grok use common asd","output":"renameauth","parse_to":"attributes","pattern":"%{COMMONAPACHELOG}","parse_from":"body"},{"type":"move","id":"renameauth","name":"rename auth","from":"attributes.auth","to":"attributes.username"}],"createdBy":"nityananda@signoz.io","createdAt":"2023-03-07T16:56:53.36071141Z"}]',
	pipelines: pipelineMockData,
	history: [
		{
			id: 'e118dedd-e996-455a-9cb2-5bf50b77fc35',
			version: 1,
			elementType: 'log_pipelines',
			createdBy: 'test2@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'DEPLOYED',
			deployResult: 'deploy successful',
			lastHash: '',
			lastConf: '',
			// eslint-disable-next-line sonarjs/no-duplicate-string
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'test2',
		},
		{
			id: '9a98673d-b5db-4281-89d3-d85ed9ffe311',
			version: 2,
			elementType: 'log_pipelines',
			createdBy: 'test3@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'DEPLOYED',
			deployResult: 'deploy successful',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'test3',
		},
		{
			id: '9fdb0813-f77f-4837-815e-bb6eedd64f68',
			version: 3,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+1@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+1',
		},
		{
			id: '87efb1cf-85b0-4aa4-934e-62a118fa4ec7',
			version: 4,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+2@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+2',
		},
		{
			id: '67ace08a-6b6c-4221-ab58-a5d3bd5eb6f2',
			version: 5,
			elementType: 'log_pipelines',
			createdBy: 'nityananda+4@signoz.io',
			active: false,
			isValid: false,
			disabled: false,
			deployStatus: 'IN_PROGRESS',
			deployResult: '',
			lastHash: '',
			lastConf: '',
			createdAt: '2021-03-07T16:56:53.36071141Z',
			createdByName: 'nityananda+4',
		},
	],
};
