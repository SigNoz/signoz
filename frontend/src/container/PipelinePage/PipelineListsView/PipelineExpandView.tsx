import { PlusCircleOutlined } from '@ant-design/icons';
import { TableLocale } from 'antd/es/table/interface';
import logEvent from 'api/common/logEvent';
import { useIsDarkMode } from 'hooks/useDarkMode';
import React, { useCallback, useMemo } from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { useTranslation } from 'react-i18next';
import {
	ActionMode,
	ActionType,
	PipelineData,
	ProcessorData,
} from 'types/api/pipeline/def';

import { tableComponents } from '../config';
import { ModalFooterTitle } from '../styles';
import { processorColumns } from './config';
import { AlertMessage } from './PipelineListsView';
import { FooterButton, StyledTable } from './styles';
import DragAction from './TableComponents/DragAction';
import ProcessorActions from './TableComponents/ProcessorActions';
import {
	getEditedDataSource,
	getProcessorUpdatedRow,
	getRecordIndex,
	getTableColumn,
} from './utils';

function PipelineExpandView({
	handleAlert,
	setActionType,
	processorEditAction,
	isActionMode,
	setShowSaveButton,
	expandedPipelineData,
	setExpandedPipelineData,
	prevPipelineData,
}: PipelineExpandViewProps): JSX.Element {
	const { t } = useTranslation(['pipeline']);
	const isDarkMode = useIsDarkMode();
	const isEditingActionMode = isActionMode === ActionMode.Editing;

	const deleteProcessorHandler = useCallback(
		(record: ProcessorData) => (): void => {
			setShowSaveButton(ActionMode.Editing);
			if (expandedPipelineData && expandedPipelineData?.config) {
				const filteredData = expandedPipelineData?.config.filter(
					(item: ProcessorData) => item.id !== record.id,
				);
				const pipelineData = { ...expandedPipelineData };
				pipelineData.config = filteredData;
				pipelineData.config.forEach((item, index) => {
					const obj = item;
					obj.orderId = index + 1;
				});
				for (let i = 0; i < pipelineData.config.length - 1; i += 1) {
					pipelineData.config[i].output = pipelineData.config[i + 1].id;
				}
				delete pipelineData.config[pipelineData.config.length - 1]?.output;
				setExpandedPipelineData(pipelineData);
			}
		},
		[expandedPipelineData, setShowSaveButton, setExpandedPipelineData],
	);

	const processorDeleteAction = useCallback(
		(record: ProcessorData) => (): void => {
			handleAlert({
				title: `${t('delete_processor')} : ${record.name}?`,
				descrition: t('delete_processor_description'),
				buttontext: t('delete'),
				onOk: deleteProcessorHandler(record),
			});
		},
		[handleAlert, deleteProcessorHandler, t],
	);

	const onSwitchProcessorChange = useCallback(
		(checked: boolean, record: ProcessorData): void => {
			if (expandedPipelineData && expandedPipelineData?.config) {
				setShowSaveButton(ActionMode.Editing);
				const findRecordIndex = getRecordIndex(
					expandedPipelineData?.config,
					record,
					'id',
				);
				const updateSwitch = {
					...expandedPipelineData?.config[findRecordIndex],
					enabled: checked,
				};
				const editedData = getEditedDataSource(
					expandedPipelineData?.config,
					record,
					'id',
					updateSwitch,
				);
				const modifiedProcessorData = { ...expandedPipelineData };
				modifiedProcessorData.config = editedData;

				setExpandedPipelineData(modifiedProcessorData);
			}
		},
		[expandedPipelineData, setExpandedPipelineData, setShowSaveButton],
	);

	const columns = useMemo(() => {
		const fieldColumns = getTableColumn(processorColumns);
		if (isEditingActionMode) {
			fieldColumns.push(
				{
					title: '',
					dataIndex: 'action',
					key: 'action',
					render: (_value, record): JSX.Element => (
						<ProcessorActions
							editAction={processorEditAction(record)}
							deleteAction={processorDeleteAction(record)}
						/>
					),
				},
				{
					title: '',
					dataIndex: 'enabled',
					key: 'enabled',
					render: (value, record) => (
						<DragAction
							isEnabled={value}
							onChange={(checked: boolean): void =>
								onSwitchProcessorChange(checked, record)
							}
						/>
					),
				},
			);
		}
		return fieldColumns;
	}, [
		isEditingActionMode,
		processorEditAction,
		processorDeleteAction,
		onSwitchProcessorChange,
	]);

	const reorderProcessorRow = useCallback(
		(updatedRow: ProcessorData[]) => (): void => {
			setShowSaveButton(ActionMode.Editing);
			if (expandedPipelineData) {
				const modifiedProcessorData = { ...expandedPipelineData };
				modifiedProcessorData.config = updatedRow;
				setExpandedPipelineData(modifiedProcessorData);
			}
		},
		[expandedPipelineData, setShowSaveButton, setExpandedPipelineData],
	);

	const onCancelReorderProcessorRow = useCallback(
		() => (): void => {
			if (expandedPipelineData) setExpandedPipelineData(expandedPipelineData);
		},
		[expandedPipelineData, setExpandedPipelineData],
	);

	const moveProcessorRow = useCallback(
		(dragIndex: number, hoverIndex: number) => {
			if (expandedPipelineData?.config && isEditingActionMode) {
				const updatedRow = getProcessorUpdatedRow(
					expandedPipelineData?.config,
					dragIndex,
					hoverIndex,
				);
				handleAlert({
					title: t('reorder_processor'),
					descrition: t('reorder_processor_description'),
					buttontext: t('reorder'),
					onOk: reorderProcessorRow(updatedRow),
					onCancel: onCancelReorderProcessorRow(),
				});
			}
		},
		[
			expandedPipelineData?.config,
			isEditingActionMode,
			handleAlert,
			t,
			reorderProcessorRow,
			onCancelReorderProcessorRow,
		],
	);

	const addNewProcessorHandler = useCallback((): void => {
		setActionType(ActionType.AddProcessor);

		logEvent('Logs: Pipelines: Clicked Add New Processor', {
			source: 'signoz-ui',
		});
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [setActionType]);

	const footer = useCallback((): JSX.Element | undefined => {
		const prevPipelinesCount = prevPipelineData?.length || 0;
		if (prevPipelinesCount === 0 || isEditingActionMode) {
			return (
				<FooterButton type="link" onClick={addNewProcessorHandler}>
					<PlusCircleOutlined />
					<ModalFooterTitle>{t('add_new_processor')}</ModalFooterTitle>
				</FooterButton>
			);
		}
		return undefined;
	}, [isEditingActionMode, prevPipelineData, addNewProcessorHandler, t]);

	const onRowHandler = (
		_data: ProcessorData,
		index?: number,
	): React.HTMLAttributes<unknown> =>
		({
			index,
			moveRow: moveProcessorRow,
		} as React.HTMLAttributes<unknown>);

	const processorData = useMemo(
		() =>
			expandedPipelineData?.config &&
			expandedPipelineData?.config.map(
				(item: ProcessorData): ProcessorData => ({
					id: item.id,
					orderId: item.orderId,
					type: item.type,
					name: item.name,
					enabled: item.enabled,
				}),
			),
		[expandedPipelineData],
	);

	const getLocales = (): TableLocale => ({
		emptyText: <span />,
	});

	return (
		<DndProvider backend={HTML5Backend}>
			<StyledTable
				locale={getLocales()}
				isDarkMode={isDarkMode}
				showHeader={false}
				columns={columns}
				rowKey="id"
				size="small"
				components={tableComponents}
				dataSource={processorData}
				pagination={false}
				onRow={onRowHandler}
				footer={footer}
			/>
		</DndProvider>
	);
}

PipelineExpandView.defaultProps = {
	expandedPipelineData: {},
};

interface PipelineExpandViewProps {
	handleAlert: (props: AlertMessage) => void;
	setActionType: (actionType?: ActionType) => void;
	processorEditAction: (record: ProcessorData) => () => void;
	isActionMode: string;
	setShowSaveButton: (actionMode: ActionMode) => void;
	expandedPipelineData?: PipelineData;
	setExpandedPipelineData: (data: PipelineData) => void;
	prevPipelineData: Array<PipelineData>;
}

export default PipelineExpandView;
