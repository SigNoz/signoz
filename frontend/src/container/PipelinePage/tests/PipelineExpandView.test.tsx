import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { pipelineMockData } from '../mocks/pipeline';
import PipelineExpandView from '../PipelineListsView/PipelineExpandView';
import { matchMedia } from './AddNewPipeline.test';

jest.mock('uplot', () => {
	const paths = {
		spline: jest.fn(),
		bars: jest.fn(),
	};
	const uplotMock = jest.fn(() => ({
		paths,
	}));
	return {
		paths,
		default: uplotMock,
	};
});

beforeAll(() => {
	matchMedia();
});

describe('PipelinePage', () => {
	it('should render PipelineExpandView section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<PipelineExpandView
							handleAlert={jest.fn()}
							setActionType={jest.fn()}
							processorEditAction={jest.fn()}
							isActionMode="viewing-mode"
							setShowSaveButton={jest.fn()}
							expandedPipelineData={pipelineMockData[0]}
							setExpandedPipelineData={jest.fn()}
							prevPipelineData={pipelineMockData}
						/>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
