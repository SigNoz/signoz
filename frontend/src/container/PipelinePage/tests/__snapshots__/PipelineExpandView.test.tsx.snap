// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage should render PipelineExpandView section 1`] = `
<DocumentFragment>
  .c2 {
  margin: 0.125rem;
  padding: 0.313rem;
  border: none;
  font-style: normal;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.25rem;
}

.c1 {
  background-color: #1668DC;
  height: 1rem;
  width: 1rem;
  font-size: 0.75rem;
  line-height: 0.813rem;
  font-weight: 400;
}

.c0 .ant-table-tbody > tr > td {
  border: none;
}

.c0 .ant-table-tbody > tr:last-child > td {
  border: none;
}

.c0 .ant-table-content {
  background: #1d1d1d;
}

<div
    class="ant-table-wrapper c0 css-dev-only-do-not-override-2i2tap"
  >
    <div
      class="ant-spin-nested-loading css-dev-only-do-not-override-2i2tap"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-small"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup>
                  <col
                    style="width: 150px;"
                  />
                </colgroup>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="grokusecommon"
                    draggable="true"
                  >
                    <td
                      class="ant-table-cell"
                      style="text-align: right;"
                    >
                      <span
                        class="ant-avatar ant-avatar-circle c1 css-dev-only-do-not-override-2i2tap"
                      >
                        <span
                          class="ant-avatar-string"
                          style="transform: scale(1) translateX(-50%);"
                        >
                          1
                        </span>
                      </span>
                    </td>
                    <td
                      class="ant-table-cell"
                      style="text-align: left;"
                    >
                      <div
                        class="c2"
                      >
                        grok use common asd
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="renameauth"
                    draggable="true"
                  >
                    <td
                      class="ant-table-cell"
                      style="text-align: right;"
                    >
                      <span
                        class="ant-avatar ant-avatar-circle c1 css-dev-only-do-not-override-2i2tap"
                      >
                        <span
                          class="ant-avatar-string"
                          style="transform: scale(1) translateX(-50%);"
                        >
                          2
                        </span>
                      </span>
                    </td>
                    <td
                      class="ant-table-cell"
                      style="text-align: left;"
                    >
                      <div
                        class="c2"
                      >
                        rename auth
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
          <div
            class="ant-table-footer"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
